### ISSUE TITLE
#include <filesystem> is not found in the system

### ISSUE THREAD

**Issue by Advaitgaur004** [Advaitgaur004/Creating_own_database#1]
can use `#include <experimental/filesystem>`
have to use `g++ main.cpp -o main -std=c++17 -lstdc++fs`
can be fixed using g++ main.cpp -o main -std=c++20

This is just for reference: @Advaitgaur004 is self-assigning the issue.

**Comment by Advaitgaur004** [comment]
It leads to the lack of data persistence in non-volatile storage. The data file is being created, but I believe the flush function is not working now.
To ensure data is written immediately, I can manually flush the stream.

Another implementation would be Binary Data Storage (Optional)
```
std::ofstream file("data.bin", std::ios::out | std::ios::binary);
int data = <some Value> ;
file.write(reinterpret_cast<char*>(&data), sizeof(data));
```

**Comment by Advaitgaur004** [comment]
Fixed #1
